addons/website_event_track/controllers/event_track.py

```diff
@@ -392,30 +392,51 @@ def event_track_proposal(self, event, **post):
 
     @http.route(['''/event/<model("event.event"):event>/track_proposal/post'''], type='http', auth="public", methods=['POST'], website=True)
     def event_track_proposal_post(self, event, **post):
-        tags = []
-        for tag in event.allowed_track_tag_ids:
-            if post.get('tag_' + str(tag.id)):
-                tags.append(tag.id)
+        if not event.can_access_from_current_website():
+            raise NotFound()
+
+        #  Only accept existing tag indices.
+        input_tag_indices = [int(tag_id) for tag_id in post['tags'].split(',') if tag_id]
+        valid_tag_indices = request.env['event.track.tag'].search([('id', 'in', input_tag_indices)]).ids
+
+        #  On the form, contact_name and at least one of contact_phone and contact_email must be filled.
+        #  Therefore, empty contacts are not considered here. No contact is created if the user do not tick
+        #  the additional information checkbox on the track proposal form.
+        contact = request.env['res.partner']
+        if 'add_contact_information' in post:
+            contact = request.env['res.partner'].sudo().create({
+                'name': post['contact_name'],
+                'email': post['contact_email'],
+                'phone': post['contact_phone'],
+            })
 
         track = request.env['event.track'].sudo().create({
             'name': post['track_name'],
             'partner_name': post['partner_name'],
-            'partner_email': post['email_from'],
-            'partner_phone': post['phone'],
-            'partner_biography': plaintext2html(post['biography']),
+            'partner_email': post['partner_email'],
+            'partner_phone': post['partner_phone'],
+            'partner_biography': plaintext2html(post['partner_biography']),
+            'partner_company_name': post['partner_company_name'],
+            'partner_function': post['partner_function'],
             'event_id': event.id,
-            'tag_ids': [(6, 0, tags)],
+            'tag_ids': [(6, 0, valid_tag_indices)],
             'user_id': False,
             'description': plaintext2html(post['description']),
-            'image': base64.b64encode(post['image'].read()) if post.get('image') else False
+            'image': base64.b64encode(post['image'].read()) if post.get('image') else False,
+            'partner_id': contact.id,
```

Comment:
Maybe order them logically in create: partner_id before partner_something fields, then contact fields .

Reply:
done